org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/MavenProjectFacade.java

```diff
@@ -634,7 +597,7 @@ public IMavenExecutionContext createExecutionContext() {
       }
       return new MavenExecutionContext(PlexusContainerManager.wrap(container, mavenProject.getClassRealm()), this);
     } catch(Exception ex) {
-      throw new RuntimeException("Aquire container failed!", ex);
+      throw new IllegalStateException("Aquire container failed!", ex);
```

Comment:
Why is this an illegal state?

Reply:
Sonar-lint complained that one should use a more specific exception than `RuntimeException`.
Since if the container creation failed, I would say that the facade is in an illegal state. Therefore IMHO `IllegalStateException` is most suitable of the existing RuntimeExceptions.

But I can revert that if that is only my opinion.

Reply:
Don't trust sonar lint... that's just useless to invent new exceptions (or use unrelated ones) that are not used just to make tools happy. Actually we once had an (checked) exception but @mickaelistria has explicitly changed that because we mostly never would be able to react than throwing an error.

 Whatever makes the container failing its not because of an illegal state (what would be for example accessing a method after an object was disposed) but either an bug in eclipse or a wrong pom configuration or whatever.

Reply:
> Don't trust sonar lint... 

In general I wouldn't say that because often the complaints/suggestions are valid and I even learned some things from that. Nevertheless they are not always right, so one should probably see that more as a suggestion than as mandatory and of course nobody is forced to follow that.

I'll revert that.